src/NumericalAlgorithms/LinearOperators/WeakDivergence.hpp

```diff
@@ -0,0 +1,191 @@
+    divergence_of_flux->initialize(fluxes.number_of_grid_points());
+  }
+
+  // Multiplies the flux be det_jac_time_inverse_jacobian.
+  const auto transform_to_logical_frame =
+      [&det_jac_times_inverse_jacobian, &fluxes](
+          auto tag_v,
+          const gsl::not_null<Variables<tmpl::list<Tags::div<FluxTags>...>>*>
+              result_buffer,
+          auto xi_or_eta_jacobian) noexcept {
+        using tag = tmpl::type_from<decltype(tag_v)>;
+        using div_tag = Tags::div<tag>;
+
+        auto& result = get<div_tag>(*result_buffer);
+        const auto& flux = get<tag>(fluxes);
+
+        for (size_t result_storage_index = 0;
+             result_storage_index < result.size(); ++result_storage_index) {
+          const auto result_tensor_index =
+              result.get_tensor_index(result_storage_index);
+          const auto x_tensor_index = prepend(result_tensor_index, 0_st);
+          auto y_tensor_index = x_tensor_index;
+          y_tensor_index[0] = 1;
+          if constexpr (Dim == 2) {
+            result[result_storage_index] =
+                get<std::decay_t<decltype(xi_or_eta_jacobian)>::value, 0>(
+                    det_jac_times_inverse_jacobian) *
+                    flux.get(x_tensor_index) +
+                get<std::decay_t<decltype(xi_or_eta_jacobian)>::value, 1>(
+                    det_jac_times_inverse_jacobian) *
+                    flux.get(y_tensor_index);
+          } else {
+            auto z_tensor_index = x_tensor_index;
+            z_tensor_index[0] = 2;
+            result[result_storage_index] =
+                get<std::decay_t<decltype(xi_or_eta_jacobian)>::value, 0>(
+                    det_jac_times_inverse_jacobian) *
+                    flux.get(x_tensor_index) +
+                get<std::decay_t<decltype(xi_or_eta_jacobian)>::value, 1>(
+                    det_jac_times_inverse_jacobian) *
+                    flux.get(y_tensor_index) +
+                get<std::decay_t<decltype(xi_or_eta_jacobian)>::value, 2>(
+                    det_jac_times_inverse_jacobian) *
+                    flux.get(z_tensor_index);
+          }
+        }
+      };
+  const auto apply_matrix_in_first_dim =
+      [](double* result, const double* const input, const Matrix& matrix,
+         const size_t size, const bool add_to_result) noexcept {
+        dgemm_<true>('N', 'N',
+                     matrix.rows(),            // rows of matrix and result
+                     size / matrix.columns(),  // columns of result and u
+                     matrix.columns(),  // columns of matrix and rows of u
+                     1.0,               // overall multiplier
+                     matrix.data(),     // matrix
+                     matrix.spacing(),  // rows of matrix including padding
+                     input,             // u
+                     matrix.columns(),  // rows of u
+                     add_to_result ? 1.0 : 0.0,  // multiplier for unused term
```

Comment:
what is meant by unused term? aren't you using it if add_to_result is true?

Reply:
relic of an old, no longer relevant comment. I'll update